package/MDAnalysis/core/groups.py

```diff
@@ -3566,21 +3609,50 @@ def unique(self):
            >>> rg2.ix
            array([0, 1, 2])
            >>> rg2.unique is rg2
-           True
+           False
 
 
         .. versionadded:: 0.16.0
         .. versionchanged:: 0.19.0 If the :class:`ResidueGroup` is already
             unique, :attr:`ResidueGroup.unique` now returns the group itself
             instead of a copy.
+        .. versionchanged:: 2.0.0
+            This function now always returns a copy.
         """
-        if self.isunique:
-            return self
         _unique = self.universe.residues[unique_int_1d(self.ix)]
-        # Since we know that _unique is a unique ResidueGroup, we set its
-        # uniqueness caches from here:
-        _unique._cache['isunique'] = True
-        _unique._cache['unique'] = _unique
+        return self._create_unique_group(_unique)
+
+    def asunique(self, sorted=False):
+        """Return a :class:`ResidueGroup` containing unique
+        :class:`Residues<Residue>` only, with optional sorting.
+
+        If the :class:`ResidueGroup` is unique, this is the group itself.
+
+        Examples
+        --------
+
+           >>> rg = u.residues[[2, 1, 2, 2, 1, 0]]
+           >>> rg
+           <ResidueGroup with 6 residues>
+           >>> rg.ix
+           array([2, 1, 2, 2, 1, 0])
+           >>> rg2 = rg.asunique()
+           >>> rg2
+           <ResidueGroup with 3 residues>
+           >>> rg2.ix
+           array([0, 1, 2])
+           >>> rg2.asunique() is rg2
+           True
+
+
+        .. versionadded:: 2.0.0
+        """
+        if self.isunique:
+            if not sorted or self.issorted:
+                return self
+        if sorted:
+            return self.unique
+        _unique = self.universe.residues[unique_int_1d_unsorted(self.ix)]
```

Comment:
This gets really repetitive but unless we want to give up different docstrings, probably can't fix it. Otherwise a `._universe_group` property would be *very* handy.